android/src/main/kotlin/com/example/flutter_gamepad/FlutterGamepadPlugin.kt

```diff
@@ -1,25 +1,91 @@
 package com.example.flutter_gamepad
 
+import android.view.KeyEvent
+import android.view.MotionEvent
+import io.flutter.embedding.android.AndroidKeyProcessor
+import io.flutter.embedding.android.AndroidTouchProcessor
+import io.flutter.view.FlutterView
+import io.flutter.embedding.engine.renderer.FlutterRenderer
+import io.flutter.embedding.engine.systemchannels.KeyEventChannel
+import io.flutter.plugin.common.EventChannel
 import io.flutter.plugin.common.MethodCall
 import io.flutter.plugin.common.MethodChannel
 import io.flutter.plugin.common.MethodChannel.MethodCallHandler
 import io.flutter.plugin.common.MethodChannel.Result
 import io.flutter.plugin.common.PluginRegistry.Registrar
+import io.flutter.plugin.editing.TextInputPlugin
+import java.lang.reflect.Field
 
-class FlutterGamepadPlugin: MethodCallHandler {
-  companion object {
-    @JvmStatic
-    fun registerWith(registrar: Registrar) {
-      val channel = MethodChannel(registrar.messenger(), "flutter_gamepad")
-      channel.setMethodCallHandler(FlutterGamepadPlugin())
+/**
+ * An extension of Flutter's AndroidTouchProcessor that delegates MotionEvents to the GamepadStreamHandler.
+ */
+class GamepadAndroidTouchProcessor(renderer: FlutterRenderer) : AndroidTouchProcessor(renderer) {
+    override fun onGenericMotionEvent(event: MotionEvent): Boolean {
+        return GamepadStreamHandler.processMotionEvent(event) || super.onGenericMotionEvent(event)
+    }
+}
+
+/**
+ * An extension of Flutter's AndroidKeyProcessor that delegates KeyEvents to the GamepadStreamHandler.
+ */
+class GamepadAndroidKeyProcessor(keyEventChannel: KeyEventChannel, textInputPlugin: TextInputPlugin) : AndroidKeyProcessor(keyEventChannel, textInputPlugin) {
+    override fun onKeyDown(keyEvent: KeyEvent) {
```

Comment:
Just curious, but why/how do gamepads trigger key events?

Reply:
Oh, on Android a gamepad button press is a kind of KeyEvent. :)

(And a gamepad thumbstick movement is a kind of MotionEvent, just like a mouse or pointer movement.)

Reply:
I added comments explaining this